samples/bidirectional-chat/csharp/Function.cs

```diff
@@ -0,0 +1,170 @@
+        private const string NewConnectionTarget = "newConnection";
+
+        [FunctionName("negotiate")]
+        public static SignalRConnectionInfo GetSignalRInfo(
+            [HttpTrigger(AuthorizationLevel.Anonymous)] HttpRequest req,
+            [SignalRConnectionInfo(HubName = Hub, UserId = "{headers.x-ms-signalr-user-id}")] SignalRConnectionInfo connectionInfo)
+        {
+            return connectionInfo;
+        }
+
+        [FunctionName("connect")]
+        public static async Task Connect([SignalRTrigger(Hub, Category.Connections, Event.Connect)]InvocationContext invocationContext,
+            [SignalR(HubName = Hub)]IAsyncCollector<SignalRMessage> signalRMessages)
+        {
+            await signalRMessages.AddAsync(new SignalRMessage()
+            {
+                Target = NewConnectionTarget,
+                Arguments = new object[] { new NewConnection(invocationContext.ConnectionId) }
+            });
+        }
+
+        [FunctionAuthorize]
+        [FunctionName("broadcast")]
+        public static async Task Broadcast([SignalRTrigger(Hub, Category.Messages, "broadcast", "message")]InvocationContext invocationContext,
+            [SignalR(HubName = Hub)]IAsyncCollector<SignalRMessage> signalRMessages,
+            string message)
+        {
+            await signalRMessages.AddAsync(new SignalRMessage()
+            {
+                Target = NewMessageTarget,
+                Arguments = new object[] { new NewMessage(invocationContext, message) }
+            });
+        }
+
+        [FunctionName("sendToGroup")]
+        public static async Task SendToGroup([SignalRTrigger(Hub, Category.Messages, "sendToGroup", "groupName", "message")]InvocationContext invocationContext,
+            [SignalR(HubName = Hub)]IAsyncCollector<SignalRMessage> signalRMessages,
+            string message, string groupName)
+        {
+            await signalRMessages.AddAsync(new SignalRMessage()
+            {
+                GroupName = groupName,
+                Target = NewMessageTarget,
+                Arguments = new object[] { new NewMessage(invocationContext, message) }
+            });
+        }
+
+        [FunctionName("sendToUser")]
+        public static async Task SendToUser([SignalRTrigger(Hub, Category.Messages, "sendToUser", "userName", "message")]InvocationContext invocationContext,
+            [SignalR(HubName = Hub)]IAsyncCollector<SignalRMessage> signalRMessages,
+            string message, string userName)
+        {
+            await signalRMessages.AddAsync(new SignalRMessage()
+            {
+                UserId = userName,
+                Target = NewMessageTarget,
+                Arguments = new[] { new NewMessage(invocationContext, message) }
+            });
+        }
+
+        [FunctionName("sendToConnection")]
+        public static async Task SendToConnection([SignalRTrigger(Hub, Category.Messages, "sendToConnection", "connectionId", "message")]InvocationContext invocationContext,
```

Comment:
How about move parameter name to be in front of each parameter. Something like this:

```cs
Task SendToConnection(..., [SignalRParameter("message")]message, [SignalRParameter("connectionId")]connectionId)
```